src/object/cli_ec.c

```diff
@@ -2553,6 +2566,7 @@ obj_ec_recov_data(struct obj_reasb_req *reasb_req, daos_obj_id_t oid,
 		sgl = &sgls[i];
 		singv = (iod->iod_type == DAOS_IOD_SINGLE);
 		stripe_sgl = &stripe_sgls[i];
+		iod->iod_size = reasb_req->orr_iods[i].iod_size;
```

Comment:
can it be only updated also in one place in obj_ec_update_iod_size()? why must set it here?

Reply:
oh, during degrade fetch, obj_ec_update_iod_size() is not enough, since degrade fetch are done through different iod/sgl.